### PR TITLE
GMOS r400 blue

### PR DESCRIPTION
**Pull request by profxj**
Extends GMOS R400 solution to the blue. 
And improves a few of the bits and pieces.

Here is the latest:

gmos_r400

Also, turns off the BPM error for non 2x2 
binning on GMOS-S.

### CODE REVIEW

File: pypeit/core/wavecal/spectrographs/templ_gemini_gmos.py [hunk 1]
```diff
@@ -11,18 +11,29 @@ def gemini_gmos_r400_hama(overwrite=False):  # GMOS R400 Hamamatsu
     binspec = 2
```
Comment (jhennawi) [inline pypeit/core/wavecal/spectrographs/templ_gemini_gmos.py:11]: we've been doing this kind of development in the Dev-Suite. I agree it should probably be done here but should we make a decision and migrate?

**Review by jhennawi — COMMENTED** [review]
Looks good.

**Comment by profxj** [comment]
Dev suite tests on GMOS :

 1 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_700 pypeit (without masters)
 2 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GN_HAM_R400_885 pypeit (without masters)
 3 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GN_HAM_NS_B600_620 pypeit (without masters)
 4 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_860 pypeit (without masters)
 4 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_860 pypeit standards (without masters)
 5 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_860 pypeit standards (without masters)
 5 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_860 pypeit_sensfunc
 6 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_860 pypeit_sensfunc
 6 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_860 pypeit_flux_setup
 7 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_860 pypeit_flux_setup
 7 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_860 pypeit_flux
 8 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_860 pypeit_flux
 8 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_860 pypeit_coadd_1dspec
 8 passed/ 1 failed/ 0 skipped FAILED  gemini_gmos/GS_HAM_R400_860 pypeit_coadd_1dspec
----
gemini_gmos/GS_HAM_R400_860 pypeit_coadd_1dspec Result: --- FAILED

Logfile:    /scratch/PypeIt-development-suite/REDUX_OUT/gemini_gmos/GS_HAM_R400_860/gemini_gmos_gs_ham_r400_860.test_1dcoadd.log
Process Id: 1342188
Start time: Mon Mar 22 13:55:32 2021
End time:   Mon Mar 22 13:55:37 2021
Duration:   0:00:05.320084
Command:    pypeit_coadd_1dspec /data/Projects/PypeIt-development-suite/coadd1d_files/gemini_gmos_gs_ham_r400_860.coadd1d

Error Messages:

End of Log:
  File "/data/Projects/Python/PypeIt/pypeit/pypmsgs.py", line 184, in error
    raise PypeItError(msg)
pypeit.pypmsgs.PypeItError: No matching objects for SPAT1602-SLIT1789-DET01.  Odds are you input the wrong OBJID




--- PYPEIT DEVELOPMENT SUITE FAILED 1/9 TESTS (Masters ignored) ---

**Comment by profxj** [comment]
Unit tests pass:

================ 256 passed, 413 warnings in 1268.58s (0:21:08) ================

**Comment by profxj** [comment]
Shenli Tang reports that 1x1 binning on GMOS-S runs smoothly on this branch.

File: pypeit/core/wavecal/spectrographs/templ_gemini_gmos.py (hunk 1 above)
Comment (profxj) [reply pypeit/core/wavecal/spectrographs/templ_gemini_gmos.py:11]: I'm not sure what you mean. 
In the DevSuite are the files for
this and not the code.

File: pypeit/core/wavecal/templates.py [hunk 2]
```diff
@@ -122,6 +125,8 @@ def build_template(in_files, slits, wv_cuts, binspec, outroot, outdir=None,
                 wv_vac, spec, pypeitFit = pypeit_arcspec(in_file, slit, binspec, binning[kk])
         else:
             wv_vac, spec = wvspec['wv_vac'], wvspec['spec']
+        # Diagnostics
+        print("wvmin, wvmax of {}: {}, {}".format(in_file, wv_vac.min(), wv_vac.max()))
```
Comment (kbwestfall) [inline pypeit/core/wavecal/templates.py]: uses `msgs.info` instead?

File: pypeit/core/wavecal/templates.py [hunk 3]
```diff
@@ -153,7 +158,10 @@ def build_template(in_files, slits, wv_cuts, binspec, outroot, outdir=None,
                     iend = np.argmin(np.abs(new_wave - wvmax))
                     # Interpolate
                     f = interp1d(wv_vac, spec)
-                    spec = f(new_wave[ipix + 1:iend])
+                    try:
+                        spec = f(new_wave[ipix + 1:iend])
+                    except:
+                        embed(header='160 of templates')
```
Comment (kbwestfall) [inline pypeit/core/wavecal/templates.py]: remove this embed?

File: pypeit/spectrographs/gemini_gmos.py [hunk 4]
```diff
@@ -507,8 +507,8 @@ def bpm(self, filename, det, shape=None, msbias=None):
 
             # Apply the mask
             xbin = int(binning.split(' ')[0])
-            if xbin != 2:
-                embed()
+            #if xbin != 2:
+            #    embed()
```
Comment (kbwestfall) [inline pypeit/spectrographs/gemini_gmos.py]: remove the comments here and on line 492?

**Review by kbwestfall — APPROVED** [review]
Looks good to me!

**Review by jhennawi — APPROVED** [review]
Approved.

File: pypeit/core/wavecal/templates.py (hunk 2 above)
Comment (profxj) [reply pypeit/core/wavecal/templates.py]: done

File: pypeit/core/wavecal/templates.py (hunk 3 above)
Comment (profxj) [reply pypeit/core/wavecal/templates.py]: done

File: pypeit/spectrographs/gemini_gmos.py (hunk 4 above)
Comment (profxj) [reply pypeit/spectrographs/gemini_gmos.py]: done

**Pull request merged**